Documentation/Plugins.md

```diff
@@ -170,11 +170,15 @@ struct MyPlugin: BuildToolPlugin {
 
 The plugin script can import *Foundation* and other standard libraries, but in the current version of SwiftPM, it cannot import other libraries.
 
+##### Build Commands
+
 In this example, the returned command is of the type `buildCommand`, so it will be incorporated into the build system's command graph and will run if any of the output files are missing or if the contents of any of the input files have changed since the last time the command ran.
 
-Note that build tool plugins are always applied to a target, which is passed in the parameter to the entry point.  Only source module targets have source files, so a plugin that iterates over source files will commonly test that the target it was given conforms to `SourceModuleTarget`.
+The target to which the plugin applies is passed as the `target` parameter.  Only source module targets have source files, so a plugin that iterates over source files will commonly test that the target it was given conforms to `SourceModuleTarget`.
+
+##### Prebuild Commands
 
-A build tool plugin can also return commands of the type `prebuildCommand`, which run before the build starts and can populate a directory with output files whose names are not known until the command runs:
+A build tool plugin can return a combination of build commands and prebuild commands.  A `prebuildCommand` runs after the build tool plugin but before the build starts.  This one populates a `GeneratedFiles/` directory:
```

Comment:
the original text said

```
Any prebuild commands are run after the plugin runs but before the build starts, and any files that are in the prebuild command's declared `outputFilesDirectory` will be evaluated as if they had been source files in the target. The prebuild command should add or remove files in this directory to reflect the results of having run the command
```
@neonichu @abertelrud which one is true?

Reply:
Is GitHub lying to me?  I do remember the words you're quoting, but the diff shows that this was the original text:

> A build tool plugin can also return commands of the type `prebuildCommand`, which run before the build starts and can populate a directory with output files whose names are not known until the command runs.

Regardless, any files whose extensions don't match known source file extensions are in fact evaluated as though they were listed as resources, which makes the text you cited wrong.

Reply:
mostly curious about `outputFilesDirectory` vs `This one populates a GeneratedFiles/`

the quote is from a paragraph down below that was deleted as part of the suggested changes

Reply:
Whatever directory the plugin passes in the `outputFilesDirectory` parameter is the one that will be checked.  There is nothing special about the name "GeneratedFiles" — that's just what the example happens to use as the name of its output directory.

Reply:
okay, then I think the  suggested documentation should point out usage of outputFilesDirectory (as done originally) and highlight that  GeneratedFiles is just an example.

Reply:
Let me see if I can understand what that implies for this PR… back soon with a proposed update.

Reply:
The original pointing out of `outputFilesDirectory` was in an inaccurate context, and disconnected from the language on line 212 that covered the same topic.  The new language that covers that information is in [this paragraph](https://github.com/apple/swift-package-manager/pull/6901/commits/370b26cd797139e8944777d56d42cc66bd9ec232).  It doesn't mention `outputFilesDirectory` but instead mentions the output files themselves.  I would expect that if the build tool left files in `outputFilesDirectory` directory that were not in the output files list it would be an error (otherwise, what's the point of having an output files list at all?)  Are you sure this needs to change?  If so, what change do you want?

Reply:
my original point was that the proposed text talks about the concrete example below (GeneratedFiles) and not about the general case (whatever is specified in outputFilesDirectory). IMO the general text should describe the functionality more generally, and not just the example. does that make sense?

maybe something like the following (combination of the new + old text):

> A build tool plugin can return a combination of build commands and prebuild commands. Any prebuild commands are run after the build tool plugin but before the build. Any files that are generate in the prebuild command's declared `outputFilesDirectory` will be evaluated as if they had been source files in the target. 
>
> In the following example, the `prebuildCommand` populates a `GeneratedFiles/` directory: